src/Workspaces/Core/Portable/Workspace/Solution/Solution.cs

```diff
@@ -975,15 +917,10 @@ public Solution RemoveAnalyzerReference(AnalyzerReference analyzerReference)
         /// <exception cref="ArgumentException"><paramref name="analyzerReferences"/> contains duplicate items.</exception>
         public Solution WithAnalyzerReferences(IEnumerable<AnalyzerReference> analyzerReferences)
         {
-            var newState = _state.WithAnalyzerReferences(
-                PublicContract.ToBoxedImmutableArrayWithDistinctNonNullItems(analyzerReferences, nameof(analyzerReferences)));
+            var collection = PublicContract.ToBoxedImmutableArrayWithDistinctNonNullItems(analyzerReferences, nameof(analyzerReferences));
 
-            if (newState == _state)
-            {
-                return this;
-            }
-
-            return new Solution(newState);
+            var newCompilationState = _compilationState.WithAnalyzerReferences(_state.WithAnalyzerReferences(collection));
```

Comment:
This one takes the state change rather than just passing it through like the other cases generally?

Reply:
yeah, this matches the AddAnalyzerReferences/REmoveAnalyzerREferences which are bit of the odd ducks.